src/listener/handlers/document-update-handler.ts

```diff
@@ -152,37 +150,40 @@ class DocumentUpdateHandler implements IHandler {
     checkEligibleForFileValidation(contentValidationReport: ValidationReport, documentUrl: string, documentType: string): Promise<ValidationReport> {
         if (documentType.toUpperCase() === 'FILE') {
             // refresh doc before checking cloudUrl
-            return this.ingestClient.retrieveMetadataDocument(documentUrl).then((doc: any) => {
-                if (doc['cloudUrl']) {
-                    // if cloud url present, proceeds with file validation
-                    if (contentValidationReport.validationState.toUpperCase() == "VALID") {
-                        return this.attemptFileValidation(contentValidationReport, doc, documentType)
+            return this.ingestClient.retrieveMetadataDocument(documentUrl)
+                .then((doc: any) => {
+                    const fileName = doc['fileName'];
+                    const fileFormat = doc['content']['file_core']['format'];
+
+                    const fileExt = fileName.substring(fileName.indexOf('.') + 1);
+
+                    if (fileFormat != fileExt && fileFormat && fileExt) {
+                        const message = `The file extension, ${fileExt}, of the file with filename, "${fileName}", doesn't match the file format, "${fileFormat}", in the metadata.`;
```

Comment:
I would put the logic that builds the validation error into its own function.

Reply:
done!